agent/xds/server.go

```diff
@@ -536,16 +536,14 @@ func (s *Server) Check(ctx context.Context, r *envoyauthz.CheckRequest) (*envoya
 
 // GRPCServer returns a server instance that can handle XDS and ext_authz
 // requests.
-func (s *Server) GRPCServer(certFile, keyFile string) (*grpc.Server, error) {
+func (s *Server) GRPCServer(tlsConfigurator *tlsutil.Configurator) (*grpc.Server, error) {
 	opts := []grpc.ServerOption{
 		grpc.MaxConcurrentStreams(2048),
 	}
-	if certFile != "" && keyFile != "" {
-		creds, err := credentials.NewServerTLSFromFile(certFile, keyFile)
-		if err != nil {
-			return nil, err
+	if tlsConfigurator != nil {
+		if cert := tlsConfigurator.Cert(); cert != nil {
+			opts = append(opts, grpc.Creds(*tlsConfigurator.TransportCredentials()))
```

Comment:
It seems a bit odd to have `TransportCredentials` return a pointer, but then dereference it for `grpc.Creds()`. 

I think TransportCredentials shouldn't return a pointer.

```suggestion
			opts = append(opts, grpc.Creds(tlsConfigurator.TransportCredentials()))
```